### PR TITLE
feat: add Kover test coverage integration

### PR DESCRIPTION
**Pull request by kirich1409**
## Summary

Integrate Kover 0.9.7 for comprehensive test coverage tracking and enforcement.

### Configuration
- ✅ Add Kover plugin to version catalog and build
- ✅ Configure HTML and XML coverage reports
- ✅ Set up coverage exclusions (generated code, test utilities)
- ✅ Add verification rules: 80% line coverage, 55% branch coverage
- ✅ Integrate coverage verification with `check` task

### Documentation
- ✅ Add coverage badge and usage guide to README
- ✅ Document coverage thresholds and commands
- ✅ Add note about manual coverage badge updates

### CI Integration
- ✅ Upload coverage reports as GitHub Actions artifacts (7-day retention)
- ✅ Add coverage summary to PR comments
- ✅ Include coverage status in GitHub step summary

### Build Configuration
- ✅ Update .gitignore to exclude coverage artifacts

### Code Quality Improvements
- ✅ Remove `endsWith` check from `NoKoinComponentInterface` to prevent false positives (e.g., NonActivity matching Activity)
- ✅ Remove unnecessary `ThreadLocal` from `MissingScopeClose` (Detekt rules run in single-threaded context)
- ✅ Add edge case test for false positive detection
- ✅ Remove tests for subclass matching (no longer supported without endsWith)

## Current Coverage

- **Line Coverage:** 94.67% (373/394 lines) ✅
- **Branch Coverage:** 59.85% (161/269 branches) ✅
- **Method Coverage:** 97.56% (40/41 methods)
- **Class Coverage:** 100% (16/16 classes)

## Test Plan

- [x] All 48 tests passing
- [x] Coverage reports generate successfully
- [x] Coverage verification passes (80% line, 55% branch)
- [x] HTML report: build/reports/kover/html/index.html
- [x] XML report: build/reports/kover/coverage.xml
- [x] CI workflow includes coverage upload
- [x] Build completes in <500ms with cache

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/main/kotlin/io/github/krozov/detekt/koin/scope/MissingScopeClose.kt [hunk 1]
```diff
@@ -26,17 +26,21 @@ internal class MissingScopeClose(config: Config) : Rule(config) {
         debt = Debt.TEN_MINS
     )
 
-    private val classesWithScopeCreation = mutableSetOf<KtClass>()
-    private val classesWithScopeClose = mutableSetOf<KtClass>()
+    // Thread-safe state using ThreadLocal to support parallel file processing
+    private val classesWithScopeCreation = ThreadLocal.withInitial { mutableSetOf<KtClass>() }
+    private val classesWithScopeClose = ThreadLocal.withInitial { mutableSetOf<KtClass>() }
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply src/main/kotlin/io/github/krozov/detekt/koin/scope/MissingScopeClose.kt]: ✅ Исправлено в коммите 3c65f9b.

Убрал ThreadLocal, вернулся к простым `mutableSetOf<KtClass>()`. Вы правы - правила Detekt обрабатывают файлы последовательно, ThreadLocal добавлял ненужную сложность и overhead без реальной пользы.

Все тесты проходят с упрощенной реализацией.

File: src/test/kotlin/io/github/krozov/detekt/koin/servicelocator/NoKoinComponentInterfaceTest.kt [hunk 2]
```diff
@@ -69,4 +69,34 @@ class NoKoinComponentInterfaceTest {
...
+            import androidx.appcompat.app.AppCompatActivity
+
+            class MainActivity : AppCompatActivity(), KoinComponent
+        """.trimIndent()
+
+        val findings = NoKoinComponentInterface(Config.empty)
+            .lint(code)
+
+        assertThat(findings).isEmpty()
+    }
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply src/test/kotlin/io/github/krozov/detekt/koin/servicelocator/NoKoinComponentInterfaceTest.kt]: ✅ Исправлено в коммите 3c65f9b.

Добавил тест для edge case `reports edge case - NonActivity should not match Activity`, который проверяет, что NonActivity правильно распознается как нарушение (не совпадает с Activity).

Также убрал тесты для FragmentActivity/AppCompatActivity, так как после удаления `endsWith` они стали некорректными - эти классы больше не распознаются автоматически.

**Comment by kirich1409** [comment]
## Исправлены все комментарии от Copilot ✅

Все 6 комментариев от code review обработаны в коммите 3c65f9b:

### Исправления кода
1. ✅ **NoKoinComponentInterface**: Убрал `endsWith` проверку - возвращен точный матчинг для предотвращения false positives
2. ✅ **MissingScopeClose**: Убрал ThreadLocal - вернулся к простым `mutableSetOf` (правила Detekt однопоточные)
3. ✅ **NoKoinComponentInterfaceTest**: Добавил edge case тест для NonActivity, убрал некорректные тесты для FragmentActivity/AppCompatActivity

### Исправления документации
4. ✅ **README.md**: Убрал ссылку из coverage badge (локальный путь не работает в GitHub)
5. ✅ **README.md**: Добавил заметку о ручном обновлении badge
6. ✅ **PR Description**: Добавил секцию "Code Quality Improvements" с описанием всех изменений в коде

### Результаты
- Все 48 тестов проходят ✅
- Все CI проверки успешны ✅  
- Покрытие: 94.67% line / 59.85% branch ✅

Готово к финальному review!
